### PR TITLE
Blocks: Extract and factorize block alignment controls

### PR DESCRIPTION
**Pull request by youknowriad**
This PR continues our Refactoring to move the controls to the `edit` function. 

**Testing instructions**

 - Block Alignment controls for the button, image, embed and table block should continue the work the way they work on master.

### CODE REVIEW

**Comment by aduth** [comment]
Is this different in purpose than #987 ? How does it relate?

**Comment by youknowriad** [comment]
Isn't #987 about text alignments? or do you think we can reuse it?

**Comment by aduth** [comment]
#987 is mainly toward a goal of letting alignment be a consideration of the block as a whole, not individual paragraphs. So seems similar in purpose to what you've proposed here, though your changes include a handful more blocks.

**Comment by youknowriad** [comment]
But don't you think we should keep distinction between TextAlignmentToolbar and BlockAlignmentToolbar ? Even if the internals are close, and could share some code, having two different components could be a good thing.

**Comment by aduth** [comment]
Hmm, what would be the practical difference between the two? Seems they only really vary in the icons they render. Both assume a callback is to be passed to specify the actual change implementation.

**Comment by youknowriad** [comment]
yes, the icons are the only differences but sometimes it's good to duplicate code for clarity. A component could have both. These are the two toolbars we have for now, but we may have other reusable toolbars a block author could use. I see them as a meaningful group of controls instead of a more generic toolbar component with maybe unrelated controls to pick. 

Anyway, this is just an idea, the generic "toggle" toolbar is a possibility too.

**Comment by aduth** [comment]
Yeah, not so opposed to the idea of duplication as much as questioning whether a distinction between text alignment vs. block alignment is really one we care to make, or is in-fact more clear.

Maybe cc @jasmussen @mtias to this point.

**Comment by youknowriad** [comment]
I'm keen to move these two (PRs #987 and this one) forward to remove the `controls` property from the block API.

File: blocks/block-alignment-toolbar/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		icon: 'align-center',
+		title: __( 'Align center' ),
+	},
+	right: {
+		icon: 'align-right',
+		title: __( 'Align right' ),
+	},
+	wide: {
+		icon: 'align-full-width',
+		title: wp.i18n.__( 'Wide width' ),
```
Comment (aduth) [inline blocks/block-alignment-toolbar/index.js]: Consistency: `__`

File: blocks/block-alignment-toolbar/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	center: {
+		icon: 'align-center',
+		title: __( 'Align center' ),
+	},
+	right: {
+		icon: 'align-right',
+		title: __( 'Align right' ),
+	},
+	wide: {
+		icon: 'align-full-width',
```
Comment (aduth) [inline blocks/block-alignment-toolbar/index.js]: I think we want `icon: 'align-wide'`

**Review by aduth — DISMISSED** [review]
Dunno if we might want to hold off on this in our push toward a stable release this week since it's not really fixing or adding anything.

**Comment by youknowriad** [comment]
Ok to postpone :)

**Pull request merged**